### PR TITLE
HHH-15160 - Adds the Postgis distance operators

### PR DESCRIPTION
**Pull request by maesenka**

### CODE REVIEW

File: documentation/src/main/asciidoc/userguide/chapters/query/spatial/Spatial.adoc [hunk 1]
```diff
@@ -110,13 +109,31 @@ relevant section.
...
+Database notes below.
+
 === Database notes
+[[spatial-configuration-dialect-postgresql]]
+Postgresql::
+
+The Postgresql dialect has support for the https://postgis.net/[Postgis spatial extension], but not the Geometric types mentioned in the
+https://www.postgresql.org/docs/current/datatype-geometric.html[Postgresql documentation].
+
+In addition tot he common spatial functions, the following functions are supported:
```
Comment (beikov) [inline documentation/src/main/asciidoc/userguide/chapters/query/spatial/Spatial.adoc:122]:
```suggestion
In addition to the common spatial functions, the following functions are supported:
```

File: hibernate-spatial/src/main/java/org/hibernate/spatial/dialect/postgis/PostgisSqmFunctionDescriptors.java [hunk 2]
```diff
@@ -7,22 +7,78 @@
...
+		//addOperator( "distance_nd_bbox", "<<#>>" );
+	}
+
+	protected void addOperator(String name, String operator) {
+		map.put(
+				FunctionKey.apply( name ),
+				new PostgisOperator(
+						name,
+						operator,
+						exactly( 2 ),
```
Comment (beikov) [inline hibernate-spatial/src/main/java/org/hibernate/spatial/dialect/postgis/PostgisSqmFunctionDescriptors.java:50]: Would be awesome if you could add a custom `ArgumentsValidator` that validates the types of the operands.

**Review by beikov — APPROVED** [review]
LGTM, just minor comments

File: hibernate-spatial/src/main/java/org/hibernate/spatial/dialect/postgis/PostgisSqmFunctionDescriptors.java (hunk 2 above)
Comment (maesenka) [reply hibernate-spatial/src/main/java/org/hibernate/spatial/dialect/postgis/PostgisSqmFunctionDescriptors.java]: I didn't opt for a custom `ArgumentsValidator` but extended the approach I saw in e.g. the avg function implementation. Could you have a look?

**Review by beikov — APPROVED** [review]

**Pull request merged**